docs/design/permissive-mode.md

```diff
@@ -0,0 +1,384 @@
+    }
+  ],
+  "components": [
+    {
+      "bom-ref": "123abc",
+      "name": "package",
+      "properties": [
+        {
+          "name": "hermeto:found_by",
+          "value": "hermeto"
+        }
+      ],
+      "purl": "pkg:golang/github.com/example/package@v1.2.3",
+      "type": "library",
+      "version": "v1.2.3"
+    },
+    {
+      "bom-ref": "123xyz",
+      "name": "package",
+
+      "properties": [
+        {
+          "name": "hermeto:found_by",
+          "value": "hermeto"
+        }
+      ],
+      "purl": "pkg:golang/github.com/another/package@v2.1.0",
+      "type": "library",
+      "version": "v2.1.0"
+    }
+  ],
+  "specVersion": "1.6",
+  "version": 1,
+  "metadata": {
+    "tools": [
+      {
+        "vendor": "red hat",
+        "name": "hermeto"
+      }
+    ]
+  }
+}
+```
+
+</details>
+
+#### Approach 2 (alternative): Properties within the components
+
+This approach involves adding properties that describe the violation directly
+to each affected component.
+
+<details>
+<summary>Example</summary>
+
+Example:
+
+```json
+{
+  "bomFormat": "CycloneDX",
+  "components": [
+    {
+      "name": "github.com/gin-gonic/gin",
+      "properties": [
+        {
+          "name": "hermeto:found_by",
+          "value": "hermeto"
+        },
+        {
+          "name": "hermeto:permissive-mode:violation",
+          "value": "gomod:vendor-directory-changed-after-vendoring"
+        }
+      ],
+      "purl": "pkg:golang/github.com/gin-gonic/gin@v1.9.1",
+      "type": "library",
+      "version": "v1.9.1"
+    },
+    {
+      "name": "github.com/gorilla/mux",
+      "properties": [
+        {
+          "name": "hermeto:found_by",
+          "value": "hermeto"
+        },
+        {
+          "name": "hermeto:permissive-mode:violation",
+          "value": "gomod:vendor-directory-changed-after-vendoring"
+        }
+      ],
+      "purl": "pkg:golang/github.com/gorilla/mux@v1.8.0",
+      "type": "library",
+      "version": "v1.8.0"
+    }
+  ],
+  "specVersion": "1.6",
+  "version": 1,
+  "metadata": {
+    "tools": [
+      {
+        "vendor": "red hat",
+        "name": "hermeto"
+      }
+    ]
+  }
+}
+```
+
+</details>
+
+### SPDX
+
+- JSON Schema: [https://github.com/spdx/spdx-spec/blob/support/2.3/schemas/spdx-schema.json](https://github.com/spdx/spdx-spec/blob/support/2.3/schemas/spdx-schema.json)
+- Validation can be done using the SPDX [online](https://tools.spdx.org/app/validate)
+tool.
+
+**NOTE:** We are using SPDX version 2.3 right now. The latest version is however,
+3.1 at the time of writing. We are not planning to upgrade to 3.1 in the near future.
+
+#### Approach 1 (preferred): Annotations within the packages
+
+This approach involves adding annotations directly to each package that describe
+permissive mode violations. Annotations are currently used for converting CycloneDX
+component properties.
+
+*This is the most practical and well-supported method for attaching custom,
```

Comment:
I am lost, which is best, per component or top-level? One part of the document says one thing, another one another. I will not block on this now, however I think it is better to refer to just one thing as the best.